service/history/handler.go

```diff
@@ -281,19 +294,16 @@ func (h *Handler) RecordActivityTaskHeartbeat(ctx context.Context, request *hist
 		return nil, h.convertError(err)
 	}
 
-	response, err2 := engine.RecordActivityTaskHeartbeat(ctx, request)
```

Comment:
There are a couple of `err2`s; I don't see why they should be here and replaced them with `err`.